docs/dg/dev/architecture/coding-conventions.md

```diff
@@ -0,0 +1,2165 @@
+    */
+  class ZipZapPlugin extends AbstractPlugin implements FooPluginInterface, BarPluginInterface {}
+```
+
+## Plugin Interface
+
+```
+[Organization]
+├── Zed
+│   ├── [Module]
+│   │    └── Communication
+│   │        └── Plugin
+│   │            └── [ConsumerModule]
+│   │                └── [PluginName]Plugin.php
+│   └── [ConsumerModule]Extension
+│       └── Dependency
+│           └── Plugin
+│               └── [PluginInterfaceName]PluginInterface.php
+│
+├── Yves
+│   ├── [Module]
+│   │   └── Plugin
+│   │       └── [ConsumerModule]
+│   │           └── [PluginName]Plugin.php
+│   └── [ConsumerModule]Extension
+│       └── Dependency
+│           └── Plugin
+│               └── [PluginInterfaceName]PluginInterface.php
+│
+├── Glue
+│   ├── [Module]
+│   │   └── Plugin
+│   │       └── [ConsumerModule]
+│   │           └── [PluginName]Plugin.php    
+│   └── [ConsumerModule]Extension
+│       └── Dependency
+│           └── Plugin
+│               └── [PluginInterfaceName]PluginInterface.php
+│
+└── Client
+    ├── [Module]        
+    │   └── Plugin
+    │       └── [ConsumerModule]
+    │           └── [PluginName]Plugin.php    
+    └── [ConsumerModule]Extension
+        └── Dependency
+            └── Plugin
+                └── [PluginInterfaceName]PluginInterface.php
+
+```
+
+**Description**
+
+- Modules which consume [Plugins](#plugin) need to define their requirements with an interface.
+- The `Plugin Interface` needs to be placed in an `extension module`.
+
+There are three modules involved:
+
+1. **Plugin definer** (aka **extension module**): The module that defines and holds the `Plugin Interface` (example: `CompanyPostCreatePluginInterface` in `CompanyExtension` module.
+2. **Plugin executor**: The module that uses the [Plugin(s)](#plugin) in its [Dependency Provider](#dependency-provider) (example: `CompanyDependencyProvider::getCompanyPostCreatePlugins()` in `Company` module)
+3. **Plugin providers**: The modules that implement a [Plugin](#plugin) (example: `CompanyBusinessUnitCreatePlugin` in `CompanyBusinessUnit` module)
+
+**Conventions**
+- `Plugin interfaces` MUST be defined in extension modules.
+  - `Extension modules` MUST be suffixed with `Extension` and follow regular module architecture.
+  - `Extension modules` MUST NOT contain anything else but `Plugin Interfaces` and [Shared application layer](#shared).
+- `Plugin Interface` MUST have a class specification with plugin-stack details.
+- Plugin interface methods MUST receive input items as collection for scalability.
+
+**Guidelines**
+
+- Operations on single items in plugin stack is forbidden, except for the following reasons:
+  - it is strictly and inevitably a single-item flow.
+  - the items go in FIFO order and there is no other way to use a collection instead.
+- Plugin interface class specification SHOULD explain:
+  - how the [Plugins](#plugin)) will be used,
+  - what are the typical use-cases of a [Plugin](#plugin).
+
```

Comment:
```suggestion
<details><summary markdown='span'>Additional Conventions for Project Development</summary>
- `Extension modules` are not mandatory for project development. 
</details>
```

Reply:
You are in Module contribution section. None of these components are related to project development.